pkg/api/store/password/store.go

```diff
@@ -0,0 +1,276 @@
+package store
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rancher/norman/types/values"
+
+	"github.com/rancher/norman/types/convert"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/types/apis/core/v1"
+	managementschema "github.com/rancher/types/apis/management.cattle.io/v3/schema"
+	projectschema "github.com/rancher/types/apis/project.cattle.io/v3/schema"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type PasswordStore struct {
+	Schemas     map[string]*types.Schema
+	Fields      map[string]map[string]interface{}
+	Stores      map[string]types.Store
+	secretStore v1.SecretInterface
+	nsStore     v1.NamespaceInterface
+}
+
+func SetPasswordStore(schemas *types.Schemas, secretStore v1.SecretInterface, nsStore v1.NamespaceInterface) {
+	modifyProjectTypes := map[string]bool{
+		"githubPipelineConfig": true,
+		"gitlabPipelineConfig": true,
+	}
+
+	pwdStore := &PasswordStore{
+		Schemas:     map[string]*types.Schema{},
+		Fields:      map[string]map[string]interface{}{},
+		Stores:      map[string]types.Store{},
+		secretStore: secretStore,
+		nsStore:     nsStore,
+	}
+
+	pwdTypes := []string{}
+
+	for _, storeType := range pwdTypes {
+		var schema *types.Schema
+		if _, ok := modifyProjectTypes[storeType]; ok {
+			schema = schemas.Schema(&projectschema.Version, storeType)
+		} else {
+			schema = schemas.Schema(&managementschema.Version, storeType)
+		}
+		data := getFields(schema, schemas)
+		id := schema.ID
+		pwdStore.Stores[id] = schema.Store
+		pwdStore.Fields[id] = data
+		schema.Store = pwdStore
+
+	}
+}
+
+func (p *PasswordStore) Create(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}) (map[string]interface{}, error) {
+	if err := p.replacePasswords(data, schema.ID); err != nil {
+	}
+	return p.Stores[schema.ID].Create(apiContext, schema, data)
+}
+
+func (p *PasswordStore) ByID(apiContext *types.APIContext, schema *types.Schema, id string) (map[string]interface{}, error) {
+	data, err := p.Stores[schema.ID].ByID(apiContext, schema, id)
+	if err != nil {
+		return nil, err
+	}
+	if err := p.assignBack(data, schema.ID); err != nil {
+		return nil, err
+	}
+	return data, err
+}
+
+func (p *PasswordStore) Context() types.StorageContext {
+	return config.ManagementStorageContext
+}
+
+func (p *PasswordStore) Delete(apiContext *types.APIContext, schema *types.Schema, id string) (map[string]interface{}, error) {
```

Comment:
curious: do we need to override the methods even if they didn't change? we don't do it for the rest of the stores, only those that need modifications, are present